src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -0,0 +1,166 @@
+        );
+    }
+
+    /**
+     * @return a copy of the specified query parameter map, with ignoreRealtimeUpdates set to true.
+     */
+    public static Map<String, String> excludeRealtime(Map<String, String> params) {
+        Map<String, String> result = new HashMap<>(params);
+        result.put(IGNORE_REALTIME_UPDATES_PARAM, "true");
+        return result;
+    }
+
+    /**
+     * Checks that, for each query provided, an itinerary exists.
+     * @return An object with a map of results and summary of itinerary existence.
+     */
+    public static ItineraryExistence checkItineraryExistence(MonitoredTrip trip, boolean checkAllDays, boolean sortDates) throws URISyntaxException {
+        // Get queries to execute by date.
+        Map<ZonedDateTime, String> queriesByDate = ItineraryUtils.getItineraryExistenceQueries(trip, checkAllDays);
+        // TODO: Consider multi-threading?
+        Map<ZonedDateTime, Itinerary> datesWithMatchingItineraries = new HashMap<>();
+
+        Collection<Map.Entry<ZonedDateTime, String>> entriesToIterate = sortDates
+            ? queriesByDate.entrySet().stream()
+                .sorted(Map.Entry.comparingByKey())
+                .collect(Collectors.toList())
+            : queriesByDate.entrySet();
+
+        for (Map.Entry<ZonedDateTime, String> entry : entriesToIterate) {
+            ZonedDateTime date = entry.getKey();
+            String params = entry.getValue();
+
+            OtpDispatcherResponse response = OtpDispatcher.sendOtpPlanRequest(params);
+            TripPlan plan = response.getResponse().plan;
+            if (plan != null && plan.itineraries != null) {
+                for (Itinerary itinerary: plan.itineraries) {
+                    // If a matching itinerary is found, save the date with the matching itinerary.
+                    // The matching itinerary will replace the original trip.itinerary.
+                    // FIXME Replace 'equals' with matching itinerary
+                    if (itinerary.equals(trip.itinerary)) {
+                        datesWithMatchingItineraries.put(date, itinerary);
+                    }
+                }
+            }
+        }
+        return new ItineraryExistence(queriesByDate.keySet(), datesWithMatchingItineraries);
+    }
+
+    /**
+     * Overload used in normal conditions.
+     */
+    public static ItineraryExistence checkItineraryExistence(MonitoredTrip trip, boolean checkAllDays) throws URISyntaxException {
+        return checkItineraryExistence(trip, checkAllDays,false);
+    }
+
+    /**
+     * Variant of checkItineraryExistence used for tests,
+     * to ensure that the order of the dates to check matches the order of the mock OTP responses for those dates.
+     */
+    public static ItineraryExistence checkItineraryExistenceOrdered(MonitoredTrip trip, boolean checkAllDays) throws URISyntaxException {
+        return checkItineraryExistence(trip, checkAllDays,true);
+    }
```

Comment:
This doesn't seem particularly helpful to make two new methods just to set a boolean argument. These could probably be removed.

Reply:
Probably the `ordered` variant doesn't need to be here. I'd rather keep the other one to hide the sorting argument, though.